src/backend/utils/adt/graphid.c

```diff
@@ -277,3 +280,38 @@ Datum graphid_hash_cmp(PG_FUNCTION_ARGS)
 
     PG_RETURN_INT32(hash);
 }
+
+int64 get_new_id(char entity_type, Oid graphoid)
+{
+    Oid seq_id;
+    int count;
+
+    // get the OID of sequence
+    if (entity_type == LABEL_KIND_VERTEX)
+    {
+        seq_id = get_relname_relid(VERTEX_ID_SEQ, graphoid);
+    }
+    else if (entity_type == LABEL_KIND_EDGE)
+    {
+        seq_id = get_relname_relid(EDGE_ID_SEQ, graphoid);
+    }
+    else
+    {
+        ereport(ERROR, errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+            errmsg(
+            "Invalid entity type value provided; expected a vertex or edge"));
+    }
+
+    for(count = EID_MIN; count <= EID_MAX; count++)
```

Comment:
We do not have support for entity ids in ag_cache right now so there is a function call missing here that would complete the sense of this loop i.e; `entity_id_exists` (similar to `label_id_exists`) and return new id only when cache does not have this id already.
so the loop would be like this
```
for(count = EID_MIN; count <= EID_MAX; count++)
    {
        int64 id;
        id = (int64) nextval_internal(seq_id, true);
        Assert(id_is_valid(id));
        if (!entity_id_exists(graphoid, id))
        {
            return (int32) id;
        }
    }
```
I implemented the initial function for getting the id because for this `entity_id_exists` we will have to work on ag_cache that I thought would be in different patch. If this work needs to be done here, let me know i will add it.

Reply:
Why do we need entity ids in ag_cache? Just curious.